File: vms-mobile/src/screens/driver/DispatchDetailScreen.js
import React, { useState, useEffect } from 'react';
import { View, ScrollView, Text, Alert, StyleSheet } from 'react-native';
import { driverApi } from '../../api/services';
import { ScreenHeader, Card, StatusBadge, InfoRow, SectionHeader, Btn } from '../../components';
import { C } from '../../utils/colors';
import dayjs from 'dayjs';

export default function DispatchDetailScreen({ route, navigation }) {
  const { id } = route.params;
  const [dispatch, setDispatch] = useState(null);
  const [starting, setStarting] = useState(false);
  const [ending,   setEnding]   = useState(false);

  useEffect(() => {
    driverApi.dispatchDetail(id).then(r => setDispatch(r.data.data)).catch(() => {});
  }, [id]);

  async function start() {
    setStarting(true);
    try {
      await driverApi.startDispatch(id);
      Alert.alert('Trip Started ▶', 'Your trip is now in progress.', [
        { text: 'Open Live Map', onPress: () => navigation.replace('ActiveTrip', { id }) },
        { text: 'OK' },
      ]);
      driverApi.dispatchDetail(id).then(r => setDispatch(r.data.data));
    } catch (e) { Alert.alert('Error', e.response?.data?.message ?? 'Failed'); }
    finally { setStarting(false); }
  }

  async function end() {
    Alert.alert('End Trip', 'Mark as completed?', [
      { text: 'Cancel' },
      { text: 'Complete', style: 'destructive', onPress: async () => {
        setEnding(true);
        try {
          await driverApi.endDispatch(id, {});
          Alert.alert('Done!', 'Trip completed.', [{ text: 'OK', onPress: () => navigation.goBack() }]);
        } catch (e) { Alert.alert('Error', e.response?.data?.message ?? 'Failed'); }
        finally { setEnding(false); }
      }},
    ]);
  }

  if (!dispatch) return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text style={{ color: C.textMuted }}>Loading…</Text>
    </View>
  );

  const canStart = ['approved','pending'].includes(dispatch.status);
  const isActive = dispatch.status === 'in_progress';

  return (
    <View style={{ flex: 1, backgroundColor: C.bg }}>
      <ScreenHeader
        title={dispatch.dispatchNo}
        subtitle="Dispatch Details"
        onBack={() => navigation.goBack()}
        right={<StatusBadge status={dispatch.status} />}
      />
      <ScrollView contentContainerStyle={{ padding: 16, paddingBottom: 120 }}>
        <Card>
          <SectionHeader title="Route" color={C.cyan} />
          <View style={{ flexDirection: 'row', gap: 10, alignItems: 'flex-start' }}>
            <View style={{ alignItems: 'center' }}>
              <View style={s.dotA} /><View style={s.line} /><View style={s.dotB} />
            </View>
            <View style={{ flex: 1 }}>
              <Text style={s.routeLabel}>Origin</Text>
              <Text style={[s.routeAddr, { color: C.primary }]}>{dispatch.origin}</Text>
              <View style={{ height: 18 }} />
              <Text style={s.routeLabel}>Destination</Text>
              <Text style={[s.routeAddr, { color: C.emerald }]}>{dispatch.destination}</Text>
            </View>
          </View>
        </Card>

        <Card>
          <SectionHeader title="Trip Info" color={C.primary} />
          <InfoRow label="Vehicle"     value={dispatch.vehicleReg} color={C.cyan} />
          <InfoRow label="Driver"      value={dispatch.driverName} />
          <InfoRow label="Date"        value={dayjs(dispatch.date).format('DD MMM YYYY')} />
          <InfoRow label="Purpose"     value={dispatch.purpose} />
          <InfoRow label="Approved By" value={dispatch.approvedBy ?? '—'} />
          {dispatch.distance && <InfoRow label="Distance" value={`${dispatch.distance} km`} color={C.primary} />}
          {dispatch.fuelUsed && <InfoRow label="Fuel Used" value={`${dispatch.fuelUsed} L`} color={C.amber} />}
        </Card>

        {(dispatch.startTime || dispatch.endTime) && (
          <Card>
            <SectionHeader title="Timing" color={C.violet} />
            <InfoRow label="Started" value={dispatch.startTime ? dayjs(dispatch.startTime).format('DD MMM HH:mm') : '—'} />
            <InfoRow label="Ended"   value={dispatch.endTime   ? dayjs(dispatch.endTime).format('DD MMM HH:mm') : '—'} />
          </Card>
        )}
      </ScrollView>

      {/* Action buttons */}
      <View style={s.footer}>
        {canStart && (
          <View style={{ flex: 1 }}>
            <Btn label="▶  Start Trip" onPress={start} loading={starting} color={C.emerald} />
          </View>
        )}
        {isActive && (
          <>
            <View style={{ flex: 1 }}>
              <Btn label="🗺  Live Map" onPress={() => navigation.navigate('ActiveTrip', { id })} color={C.primary} />
            </View>
            <View style={{ flex: 1 }}>
              <Btn label="■  End Trip" onPress={end} loading={ending} color={C.rose} />
            </View>
          </>
        )}
      </View>
    </View>
  );
}

const s = StyleSheet.create({
  dotA:       { width: 11, height: 11, borderRadius: 6, backgroundColor: C.primary, borderWidth: 2, borderColor: C.primaryBg },
  dotB:       { width: 11, height: 11, borderRadius: 3, backgroundColor: C.emerald, borderWidth: 2, borderColor: C.emeraldBg },
  line:       { width: 2, flex: 1, minHeight: 30, backgroundColor: C.primary, opacity: 0.25, marginVertical: 4 },
  routeLabel: { fontSize: 11, color: C.textMuted, fontWeight: '600', textTransform: 'uppercase', letterSpacing: 0.5 },
  routeAddr:  { fontSize: 15, fontWeight: '700', marginTop: 2 },
  footer:     { flexDirection: 'row', gap: 10, padding: 16, backgroundColor: C.white,
    borderTopWidth: 1, borderTopColor: C.borderLight },
});

File: vms_flutter/lib/screens/driver/dispatch_detail_screen.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../../core/api.dart';
import '../../core/constants.dart';
import '../../models/models.dart';
import '../../widgets/widgets.dart';
import 'active_trip_screen.dart';

class DispatchDetailScreen extends StatefulWidget {
  final int id;
  const DispatchDetailScreen({super.key, required this.id});
  @override State<DispatchDetailScreen> createState() => _DispatchDetailScreenState();
}

class _DispatchDetailScreenState extends State<DispatchDetailScreen> {
  Dispatch? _d;
  bool _loading = true, _starting = false, _ending = false;

  @override void initState() { super.initState(); _load(); }

  Future<void> _load() async {
    try {
      final res = await ApiClient.get('/driver/dispatches/${widget.id}');
      setState(() => _d = Dispatch.fromJson(res['data'] as Map<String,dynamic>));
    } catch (_) {}
    setState(() => _loading = false);
  }

  String _fmt(String? dt) {
    if (dt == null) return '—';
    try { return DateFormat('dd MMM yyyy, HH:mm').format(DateTime.parse(dt)); }
    catch (_) { return dt; }
  }

  Future<void> _start() async {
    setState(() => _starting = true);
    try {
      await ApiClient.post('/driver/dispatches/${_d!.id}/start', {});
      await _load();
      if (!mounted) return;
      final goLive = await showDialog<bool>(context: context,
        builder: (_) => AlertDialog(title: const Text('Trip Started ▶'),
          content: const Text('Trip is now in progress. Open live map?'),
          actions: [
            TextButton(onPressed: () => Navigator.pop(context, false), child: const Text('Later')),
            TextButton(onPressed: () => Navigator.pop(context, true),
              child: const Text('Open Map', style: TextStyle(color: AppColors.primary))),
          ]));
      if (goLive == true && mounted && _d != null) {
        Navigator.push(context, MaterialPageRoute(
          builder: (_) => ActiveTripScreen(dispatch: _d!)));
      }
    } catch (e) {
      if (mounted) ScaffoldMessenger.of(context)
        .showSnackBar(SnackBar(content: Text(e.toString())));
    } finally { if (mounted) setState(() => _starting = false); }
  }

  Future<void> _end() async {
    final ok = await showDialog<bool>(context: context,
      builder: (_) => AlertDialog(title: const Text('End Trip'),
        content: const Text('Mark as completed?'),
        actions: [
          TextButton(onPressed: () => Navigator.pop(context, false), child: const Text('Cancel')),
          TextButton(onPressed: () => Navigator.pop(context, true),
            child: const Text('Complete', style: TextStyle(color: AppColors.rose))),
        ]));
    if (ok != true) return;
    setState(() => _ending = true);
    try {
      await ApiClient.post('/driver/dispatches/${_d!.id}/end', {});
      if (mounted) Navigator.pop(context);
    } catch (e) {
      if (mounted) ScaffoldMessenger.of(context)
        .showSnackBar(SnackBar(content: Text(e.toString())));
    } finally { if (mounted) setState(() => _ending = false); }
  }

  @override
  Widget build(BuildContext context) {
    final canStart = _d?.status == 'approved' || _d?.status == 'pending';
    final isActive = _d?.status == 'in_progress';

    return Scaffold(
      backgroundColor: AppColors.bg,
      appBar: AppBar(title: Text(_d?.dispatchNo ?? 'Trip Detail'),
        actions: [if (_d != null) Padding(padding: const EdgeInsets.only(right: 16),
          child: StatusBadge(_d!.status))]),
      body: _loading
        ? const Center(child: CircularProgressIndicator())
        : _d == null ? const EmptyState(emoji: '❌', title: 'Not found')
        : ListView(padding: const EdgeInsets.all(16), children: [
          AppCard(child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
            const SectionLabel('Route', color: AppColors.cyan),
            RouteConnector(from: _d!.origin, to: _d!.destination),
          ])),
          AppCard(child: Column(children: [
            const SectionLabel('Trip Details'),
            InfoRow(label: 'Vehicle',     value: _d!.vehicleReg, valueColor: AppColors.cyan),
            InfoRow(label: 'Driver',      value: _d!.driverName),
            InfoRow(label: 'Date',        value: _fmt(_d!.date)),
            if (_d!.purpose != null) InfoRow(label: 'Purpose', value: _d!.purpose!),
            if (_d!.approvedBy != null) InfoRow(label: 'Approved By', value: _d!.approvedBy!),
            if (_d!.distance != null) InfoRow(label: 'Distance', value: '${_d!.distance} km',
              valueColor: AppColors.primary),
          ])),
          if (_d!.startTime != null || _d!.endTime != null)
            AppCard(child: Column(children: [
              const SectionLabel('Timing', color: AppColors.violet),
              InfoRow(label: 'Started', value: _fmt(_d!.startTime)),
              InfoRow(label: 'Ended',   value: _fmt(_d!.endTime)),
            ])),
          const SizedBox(height: 12),
          if (canStart) GradientButton(label: '▶   Start Trip', icon: Icons.play_arrow,
            colors: const [AppColors.emerald, Color(0xFF059669)],
            onPressed: _start, loading: _starting),
          if (isActive) ...[
            const SizedBox(height: 10),
            GradientButton(label: '🗺   Open Live Map', icon: Icons.map_outlined,
              onPressed: () => Navigator.push(context, MaterialPageRoute(
                builder: (_) => ActiveTripScreen(dispatch: _d!)))),
            const SizedBox(height: 10),
            GradientButton(label: '■   End Trip',
              colors: const [AppColors.rose, Color(0xFFE11D48)],
              onPressed: _end, loading: _ending),
          ],
          const SizedBox(height: 40),
        ]),
    );
  }
}
